bolt/lib/Passes/BinaryPasses.cpp

```diff
@@ -1553,36 +1553,28 @@ Error PrintProgramStats::runOnFunctions(BinaryContext &BC) {
     const bool Ascending =
         opts::DynoStatsSortOrderOpt == opts::DynoStatsSortOrder::Ascending;
 
-    if (SortAll) {
-      llvm::stable_sort(Functions,
-                        [Ascending, &Stats](const BinaryFunction *A,
-                                            const BinaryFunction *B) {
-                          auto StatsItr = Stats.find(A);
-                          assert(StatsItr != Stats.end());
-                          const DynoStats &StatsA = StatsItr->second;
-
-                          StatsItr = Stats.find(B);
-                          assert(StatsItr != Stats.end());
-                          const DynoStats &StatsB = StatsItr->second;
-
-                          return Ascending ? StatsA < StatsB : StatsB < StatsA;
-                        });
-    } else {
-      llvm::stable_sort(
-          Functions, [Ascending, &Stats](const BinaryFunction *A,
-                                         const BinaryFunction *B) {
-            auto StatsItr = Stats.find(A);
-            assert(StatsItr != Stats.end());
-            const DynoStats &StatsA = StatsItr->second;
-
-            StatsItr = Stats.find(B);
-            assert(StatsItr != Stats.end());
-            const DynoStats &StatsB = StatsItr->second;
-
-            return Ascending ? StatsA.lessThan(StatsB, opts::PrintSortedBy)
-                             : StatsB.lessThan(StatsA, opts::PrintSortedBy);
-          });
-    }
+    std::function<bool(const DynoStats &, const DynoStats &)>
+        DynoStatsComparator =
+            SortAll ? [](const DynoStats &StatsA,
```

Comment:
Please sink SortAll down further

Reply:
By sink - do you mean remove some tabs?

Reply:
Sorry, I meant moving it down into lambda and using it as part of `return` statement

Reply:
I'm interpreting sinking SortAll as also sinking the ternary operator into the return statement such that we return either StatsA < StatsB or StatsA.lessThan(StatsB, opts::PrintSortedBy) conditioned on SortAll. However, my thinking is that with my current implementation we'll only have to check SortAll once, instead of at every comparison, which would be marginally more efficient. Did you mean something different by sinking the SortAll or was my thinking correct and your reasoning more for code brevity?

Reply:
Yes, checking once is perhaps more efficient, whereas I was seeking simpler code overall.